File: post.py
from flask import Flask, render_template, request, jsonify, Blueprint
from pymongo import MongoClient
from random import randint
import certifi


app = Flask(__name__)

import jwt
import datetime
import hashlib
from random import randint

SECRET_KEY = 'SPARTAAAAA!!!'

bp = Blueprint('post', __name__, url_prefix='/')
@bp.route('/writeDiary')
def write_diary():
    token_receive = request.cookies.get('mytoken')
    try:
        payload = jwt.decode(token_receive, SECRET_KEY, algorithms=['HS256'])
        user_info = db.users.find_one({"userId": payload['userId']})
        return render_template('postDiary.html', username=user_info["username"])
    except (jwt.ExpiredSignatureError, jwt.exceptions.DecodeError):
        return render_template('postDiary.html')

@bp.route('/postDiary', methods=['POST'])
def post_diary():
    title_receive = request.form['title_give']
    content_receive = request.form['content_give']
    date_receive = request.form['date_give']
    emoticon_receive = request.form['emoticon_give']
    username_receive = request.form['username_give']
    userId_receive = request.form['userId_give']

    countId = list(db.contents.find({},{'_id':False}))
    contentId = len(countId) + randint(0,1000)

    doc = {
        'title' : title_receive,
        'content' : content_receive,
        'date' : date_receive,
        'contentId' : contentId,
        'emoticon' : emoticon_receive,
        'username' : username_receive,
        'userId': userId_receive
    }
    db.contents.insert_one(doc)

    return jsonify({'msg' : '일기가 저장되었습니다.'})

File: static/js/post.js
$.ajax({
    type: 'GET',
    url: '/usertoken',
    data: {},
    async: false,
    success: function (response) {
        console.log(response['username']);
        console.log(response['userId']);
        $('#userId').val(response['userId'])
        $('#username').val(response['username'])
    }
});

function postDiary() {
    let check_content = document.getElementById("content");
    let title = $('#title').val()
    // let content = $('#content').val().replace(/\n/g, "<br>")
    let content = $('#content').val()
    let emoticon = $('input[name="chk_info"]:checked').val()
    let userId = $('#userId').val()
    let username = $('#username').val()
    // let username = document.getElementById('username').innerText
    let date = new Date();

    if (check_content.value == "") {
        alert("내용을 입력해주세요.");
        return false;
    } else {
        check_contents()
        $.ajax({
            type: 'POST',
            url: '/postDiary',
            data: {
                title_give: title,
                content_give: content,
                date_give: Date,
                userId_give: userId,
                emoticon_give: emoticon,
                username_give: username
            },

            success: function (response) {
                alert(response['msg'])
                window.location.href = '/'

            }
        });
    }
}


function check_contents() {
    let check_content = document.getElementById("content");
    if (check_content.value == "") {
        alert("내용을 입력해주세요.");
        return false;
    }
}



